exporter/common.go

```diff
@@ -91,42 +91,93 @@ func makeExcludeFilter(exclude []string) *primitive.E {
 	return &primitive.E{Key: "$and", Value: filterExpressions}
 }
 
-// makeDBsFilter creates a filter to list all databases or only the ones in the specified
-// namespaces list. Namespaces have the form of <db>.<collection> and the collection name
-// can have a dot. Example: db1.collection.one -> db: db1, collection: collection.one
-// db1, db2.col2, db3.col.one will produce [db1, db2, db3].
 func makeDBsFilter(filterInNamespaces []string) *primitive.E {
-	dbs := []string{}
-	if len(dbs) == 0 {
+	filterExpressions := []bson.D{}
+
+	nss := removeEmptyStrings(filterInNamespaces)
+	for _, namespace := range nss {
+		parts := strings.Split(namespace, ".")
+		filterExpressions = append(filterExpressions,
+			bson.D{{Key: "name", Value: bson.D{{Key: "$eq", Value: parts[0]}}}},
+		)
+	}
+
+	if len(filterExpressions) == 0 {
 		return nil
 	}
 
-	for _, namespace := range filterInNamespaces {
-		parts := strings.Split(namespace, ".")
-		dbs = append(dbs, parts[0])
+	return &primitive.E{Key: "$or", Value: filterExpressions}
+}
+
+func removeEmptyStrings(items []string) []string {
+	cleanList := []string{}
+
+	for _, item := range items {
+		if item == "" {
+			continue
+		}
+		cleanList = append(cleanList, item)
+	}
+
+	return cleanList
+}
+
+func unique(slice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+
+	for _, entry := range slice {
+		if _, value := keys[entry]; !value {
```

Comment:
Do you intend to use `value` instead of the idiomatic `ok` here? `value := keys[entry]; !value {` looks simpler here